examples/gno.land/r/docs/commondao/commondao_proposals.gno

```diff
@@ -0,0 +1,107 @@
+package commondao
+
+import (
+	"errors"
+	"std"
+	"time"
+
+	"gno.land/p/nt/commondao"
+)
+
+// Proposal implements the commondao.Proposal interface
+// It contains anything useful to you in a proposal and defines how the proposal behaves
+type proposal struct {
+	TitleProp       string
+	DescriptionProp string
+	CallerProp      std.Address
+}
+
+// Title returns the title of the proposal
+func (p *proposal) Title() string {
+	return p.TitleProp
+}
+
+// Body returns the full description of the proposal
+func (p *proposal) Body() string {
+	return p.DescriptionProp
+}
+
+// VotingPeriod defines the length of time the proposal is open for voting
+// Another possible implementation would be to have a constructor
+// So that each proposal can implement its own duration and tally
+func (c *proposal) VotingPeriod() time.Duration {
+	return 24 * time.Hour // Voting period is 24 hours here
+}
+
+// Tally determines whether the proposal passes based on voting results
+// There are multiple ways to tally votes, pick the one most suited to your use case
+func (c *proposal) Tally(r commondao.ReadonlyVotingRecord, m commondao.MemberSet) (bool, error) {
+	yesCount := r.VoteCount(commondao.ChoiceYes)
+
+	memberCount := DAO.Members().Size()
+	fivePercent := (memberCount*5 + 99) / 100 // fivePercent calculate what is 5% of the DAO's member
+
+	return yesCount >= fivePercent, nil // The proposal is considered as "passed" as soon as 5% of members voted yes
+}
+
+// Execute will execute what you decide once the proposal is considered passed
+func (p *proposal) Execute(cur realm) error {
+	// Here the proposal executed is the addition of the creator of the proposal to the DAO
+	if DAO.Members().Has(p.CallerProp) {
+		return errors.New(p.CallerProp.String() + " is already a DAO member")
+	}
+
+	DAO.Members().Add(p.CallerProp)
+	return nil
+}
+
+// CreateProposal constructs and submits a new proposal to the DAO
+// The proposal is stored in the ActiveProposal storage
+func CreateProposal(cur realm, title, desc string) {
+	caller := std.PreviousRealm().Address()
+
+	// Set the new proposal's title and description
+	prop := &proposal{
+		TitleProp:       title,
+		DescriptionProp: desc,
+		CallerProp:      caller,
+	}
+
+	DAO.MustPropose(caller, prop) // Panics if the creation of the proposal fails, use Propose() if you don't want the possible panic
+}
```

Comment:
Optionally, it's also possible to hard code the proposal title to better describe the intention of the proposal and to avoid callers to use random titles, by changing to something like for example:

```suggestion
func CreateProposal(cur realm, desc string) {
	caller := std.PreviousRealm().Address()

	// Create a new proposal with a description and new member address
	prop := &proposal{
		TitleProp:       "Add DAO Member "+caller.String(),
		DescriptionProp: desc,
		CallerProp:      caller,
	}

	DAO.MustPropose(caller, prop) // Panics if the creation of the proposal fails, use Propose() if you don't want the possible panic
}
```